ansible_base/jwt_consumer/common/auth.py

```diff
@@ -258,17 +260,89 @@ def get_role_definition(self, name: str) -> Optional[Model]:
 
     def process_rbac_permissions(self):
         """
-        This is a default process_permissions which should be usable if you are using RBAC from DAB
+        Process RBAC permissions using claims hash logic
         """
         if self.token is None or self.user is None:
-            logger.error("Unable to process rbac permissions because user or token is not defined, please call authenticate first")
+            logger.error("Unable to process rbac permissions because user or token is not defined")
             return
 
+        jwt_claims_hash = self.token.get("claims_hash")
+        if not jwt_claims_hash:
+            logger.error("No claims_hash found in JWT token")
+            return
+
+        user_ansible_id = self.token.get("sub")
+        if not user_ansible_id:
+            logger.error("No subject (sub) found in JWT token")
+            return
+
+        # Check cached claims hash
+        cached_claims_hash = self.cache.get_cached_claims_hash(user_ansible_id)
```

Comment:
I was just did a quick look and I don't think we ever validate that sub is a valid ansible_id anywhere. We likely don't care here but we should probablly do that somewhere to prevent getting a common value like "None" (as a string), which would make us care here since everyones permissions would effectivly end up as whomever last logged in. Thoughts?